sqtom/fitting_1d.py

```diff
@@ -26,18 +26,77 @@
 
 
 import numpy as np
+from scipy.optimize import root_scalar
 from lmfit import Minimizer, Parameters
 from sqtom.forward_solver import degenerate_pmf
 
 
+def two_schmidt_mode_guess(pd_data, sq_label="sq_", noise_fraction=0.001):
+    """Given a single mode histogram, this function generates a "physically" motivated guess for the loss, Schmidt occupations
+    and dark counts parameters.
+
+    This model is sensible only if the average g2 is above 2.
+
+    Args:
+        pd_data (array): rectangular array with the probability mass functions of the photon events
+        sq_label (string): label for the squeezing parameters.
+
+    Returns:
+        dict: dictionary containing a set of "reasonable" model parameters
+    """
+    res = marginal_calcs_1d(pd_data)
+    nmean = res["n"]
+    g2 = res["g2"]
+    P0 = pd_data[0]
+
+    def findeta(eta, nmean, g2, P0):
+        a = nmean / 4
+        b = (3 - g2) * (nmean ** 2) / 4 - nmean
+        c = (g2 - 3) * nmean ** 2
+        d = (3 - g2) * nmean ** 2 + 2 * nmean + 1 - 1 / P0 ** 2
+        return a * eta ** 3 + b * eta ** 2 + c * eta + d
+
+    eta_set = np.linspace(-0.01, 1.01, num=52)
+    function_root_search = np.array([findeta(i, nmean, g2, P0) for i in eta_set])
+    indices = np.array([])
+    for i in range(function_root_search.size - 1):
+        if function_root_search[i + 1] / function_root_search[i] < 0:
+            indices = np.append(indices, i)
+            indices = np.append(indices, i + 1)
+    eta = root_scalar(
+        findeta,
+        args=(nmean, g2, P0),
+        bracket=(eta_set[int(indices[-1] - 1)], eta_set[int(indices[-1])]),
+    ).root
+    if eta > 1:
+        eta = 1
+    if eta < 0:
+        eta = 0
+    if (g2 - 2) * nmean ** 2 - eta * nmean >= 0 and eta > 0:
```

Comment:
the code would be easier to read if you define `d = (g2 - 2) * nmean ** 2 - eta * nmean` and then used this variable for the `if` statement and also for the calculation of `n0` and `n1`.